server/base/src/main/java/org/apache/accumulo/server/constraints/MetadataConstraints.java

```diff
@@ -128,6 +129,23 @@ private static ArrayList<Short> addIfNotPresent(ArrayList<Short> lst, int intVio
     return lst;
   }
 
+  /*
+   * Validates the data file metadata by creating a StoredTabletFile for the metadata string that
+   * will eagerly load the contained ReferencedTabletFile which performs validation.
+   *
+   * We have to catch a generic Exception as various errors can occur including
+   * NullPointerException, IllegalArgumentException, etc.
+   */
+  private static ArrayList<Short> validateDataFilePath(ArrayList<Short> violations,
+      String metadata) {
+    try {
+      new StoredTabletFile(metadata);
```

Comment:
So that would be true if this was staying a path but it won't for long. In the future this metadata won't be a path, it will be json with a path and range with the changes with #3480. So we we would need to have another utility to validate or still use StoredTabletFile so it can read and parse the json

Reply:
If there's an opportunity to simplify something now, even if it changes later, it might still be worth doing, so the subsequent changes are modifying a simpler implementation. That's better for code reviews, and to see what is actually being accomplished in the subsequent changes. It can also help avoid technical debt, because sometimes when we have more complex code, and we improve a portion of it, we leave behind layers from the original implementation that aren't needed anymore. Simplifying now, whenever possible, helps with that.

That said, I don't know how much this would really simplify things here (haven't looked).

Reply:
It may simplify it a little but not really that much, it's essentially one line of code either way, we either pass the metadata to StoredTabletFile to parse it or pass it to parsePath().  I don't know that it is worth it as I think that change actually makes it more error prone. 

The idea here is we are validating the metadata for a data file. While currently metadata is a path, it doesn't have to be, and if it changes later things break at runtime with no warning ahead of time. If we keep this as StoredTabletFile it is future proof...no matter what happens to the format of the metadata this will always work because StoredTabletFile will know how to handle it. 

If we wanted to try and avoid the construction/GC of an object we could try and create a utility method that can parse/handle metadata. I am thinking maybe I could create a static method on StoredTabletFile to handle the validation so it's a little more clear and maybe we can avoid the object creation. So then it just becomes something like:

```java
try {
  StoredTabletFile.validate(metadata);
} catch (Exception e) {
  violations = addViolation(violations, 9);
}
```

Reply:
I actually went to try and add a static validate and noticed there was an existing validate method that isn't used anywhere going back to 2.1 so I created a PR to remove that first: https://github.com/apache/accumulo/pull/3540

Also, the one downside if we do the validation somewhere else (like a static validate method) vs the constructor is potentially an issue with different validation being done in different code paths (constructor could end up getting out of sync with the static method) but hopefully if someone is changing the StoredTabletFile class they would make sure that doesn't happen.

Reply:
#3540 is merged so I can look at creating a validate method and updating this PR.